examples/ex1.cpp

```diff
@@ -144,7 +153,8 @@ int main(int argc, char *argv[])
    //    corresponding to the Laplacian operator -Delta, by adding the Diffusion
    //    domain integrator.
    BilinearForm *a = new BilinearForm(fespace);
-   a->AddDomainIntegrator(new DiffusionIntegrator(one));
+   if (pa) { a->AddDomainIntegrator(new PADiffusionIntegrator(one)); }
```

Comment:
I think we should try to simplify the design, so that the definition of the `BilineaerForm` (in terms of its integrators) is independent of what type of assemply/evaluation will be used.

Reply:
We could also consider a design like:
```
VariationalForm vf = DiffusionIntegrator(one) + dgtraceIntegrator(coeff);
FiniteElementOperator feOp = vf.getFEOperator( fes, PARTIAL );
```

Reply:
@YohannDudouit I personally like this modern C++ style, but it's a very different feel from the rest of MFEM. This interface with `vf.getPAOperator(fes)`, `vf.getHypreSparseMatrixOperator(fes)`, etc. is really expressive, but I'm not sure how DOF elimination would fit in.